metadata/src/main/java/org/apache/kafka/controller/QuorumFeatures.java

```diff
@@ -63,7 +63,7 @@ public static Map<String, VersionRange> defaultFeatureMap(boolean enableUnstable
                     MetadataVersion.latestTesting().featureLevel() :
                     MetadataVersion.latestProduction().featureLevel()));
         for (Feature feature : Feature.PRODUCTION_FEATURES) {
-            short maxVersion = enableUnstable ? feature.latestTesting() : feature.latestProduction();
+            short maxVersion = enableUnstable ? feature.latestTesting() : feature.defaultLevel(MetadataVersion.LATEST_PRODUCTION);
```

Comment:
@junrao @dongnuo123 I noticed we didn't change the defaults here on the previous PR. I have done so here. A test was failing since the production version for transaction version is now not the same as the default based on the latest production MV.

Reply:
@jolshan : Not sure that I understand this change. The result of `defaultFeatureMap` is used for Controller/Broker registration. So, it seems that we should pass in the max supported version of each feature, instead of the default version, right? In fact, `defaultFeatureMap` should be renamed to sth like `supportedFeatureMap`.

> A test was failing since the production version for transaction version is now not the same as the default based on the latest production MV.

Hmm, I thought that with https://github.com/apache/kafka/pull/17886, it's ok for the latest production version for TV to be different from the default. It just needs to be larger.

Reply:
Updated with the changes to the name and the test